src/inversion_scripts/operators/TROPOMI_operator.py

```diff
@@ -529,7 +529,16 @@ def read_tropomi(filename):
         with xr.open_dataset(filename, group="PRODUCT/SUPPORT_DATA/INPUT_DATA") as tropomi_data:
             dat["methane_profile_apriori"] = tropomi_data["methane_profile_apriori"].values[0, :, :, ::-1]  # mol m-2
             dat["dry_air_subcolumns"] = tropomi_data["dry_air_subcolumns"].values[0, :, :, ::-1]  # mol m-2
-            dat["surface_classification"] = (tropomi_data["surface_classification"].values[0, :, :].astype("uint8") & 0x03).astype(int)
+
+            # Surface classification values of NaN will be filtered out
+            # due to a low QA value. We can make sure by setting them to 5
+            # and making sure nothing outside of [0,1,2,3] makes it through.
+            sc = tropomi_data["surface_classification"].values[0, :, :]
+            nan_mask = np.isnan(sc)
+            sc_no_nans = np.nan_to_num(sc, nan=0)
+            sc = (sc_no_nans.astype("uint8") & 0x03).astype(int)
+            sc[nan_mask] = 5
```

Comment:
maybe I am missing something, but is there a reason why `nan_mask` exists? Couldn't you do:
```
sc = tropomi_data["surface_classification"].values[0, :, :]
sc_no_nans = np.nan_to_num(sc, nan=5)
sc = (sc_no_nans.astype("uint8") & 0x03).astype(int)
dat["surface_classification"] = sc
```
to the same effect?

Reply:
Hi Lucas -- it doesn't matter since these will be filtered out anyway, but this would be cast to 1 which would prevent a potential error check on if your filtered results contain anything that is not [0,1,2,3].

```
import numpy as np
(np.array((5)).astype("uint8") & 0x03).astype(int)
```
> 1